tools/chplvis/ConcurrencyView.cxx

```diff
@@ -0,0 +1,425 @@
+/*
+ * Copyright 2016 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdio.h>
+#include <assert.h>
+
+#include "ConcurrencyView.h"
+#include "chplvis.h"
+
+#include "SubView.h"
+#include "SelectBrowser.h"
+
+#include <FL/fl_draw.H>
+#include <FL/fl_ask.H>
+
+// Callback for this view
+
+void backCallback (Fl_Widget *w, void *p) {
+  ConcurrencyView *view = (ConcurrencyView *)w->parent();
+  view->showTaskBox();
+  //printf ("backButtonClicked\n");
```

Comment:
I tend to leave debugging printf() in the code until I'm sure I won't need them, but haven't done a debug framework because I typically need to debug some small section of code and turning all the debugging code would make it very difficult to find the one I need.   I could just go clean up the debugging prints in all the files, if you prefer that.

And yes, this is a really big PR.   Once I got to the point where I was changing from the window based UI to the view based UI, there was no real intermediate point that felt like a good point to merge.   This is really the first version since the end of July that is coherent and doesn't have a bunch of problems and missing features.

Code could be cleaned up a lot but I ran out of time.